Tinder.API/Hubs/ChatHub.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Text.Json;
+using AutoMapper;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Caching.Distributed;
+using Tinder.API.DTO.CreateDto;
+using Tinder.API.Hubs.Connection;
+using Tinder.API.Hubs.Interfaces;
+using Tinder.Bll.Exceptions;
+using Tinder.BLL.Exceptions;
+using Tinder.BLL.Interfaces;
+using Tinder.BLL.Models;
+
+namespace Tinder.API.Hubs
+{
+    public class ChatHub : Hub<IChatHub>
+    {
+
+        private readonly IChatService _chatService;
+        private readonly IUserService _userService;
+        private readonly IMessageService _messageService;
+        private readonly IDistributedCache _cache;
+        private readonly IMapper _mapper;
+
+        public ChatHub(IChatService chatService, IUserService userService, IMessageService messageService, IDistributedCache cache, IMapper mapper)
+        {
+            _chatService = chatService;
+            _userService = userService;
+            _messageService = messageService;
+            _cache = cache;
+            _mapper = mapper;
+        }
+
+        public async Task JoinChat(ChatConnection connection)
+        {
+            var chat = await _chatService.GetByIdAsync(connection.ChatId, cancellationToken: default) ?? throw new NotFoundException("Chat is not found");
+            var user = await _userService.GetByIdAsync(connection.UserId, cancellationToken: default) ?? throw new NotFoundException("User is not found");
+            
+            if (chat.Users.All(u => u.Id != user.Id)) throw new BadRequestException("User is not allowed to be in the chat");

```

Comment:
may be use !Any() here?

Reply:
which one is better to use or it doesn't really matter

Reply:
Doesn't matter